File: src/NodeParser/AnyTypeNodeParser.ts
import ts from "typescript";
import type { Context } from "../NodeParser.js";
import type { SubNodeParser } from "../SubNodeParser.js";
import { AnyType } from "../Type/AnyType.js";
import type { BaseType } from "../Type/BaseType.js";

export class AnyTypeNodeParser implements SubNodeParser {
    public supportsNode(node: ts.KeywordTypeNode): boolean {
        return node.kind === ts.SyntaxKind.AnyKeyword || node.kind === ts.SyntaxKind.SymbolKeyword;
    }
    public createType(node: ts.KeywordTypeNode, context: Context): BaseType {
        return new AnyType();
    }
}

File: internal/parser/any_type_node_parser.go
package parser

import (
	"github.com/microsoft/typescript-go/shim/ast"

	"github.com/vega/ts-json-schema-generator/internal/types"
)

// AnyTypeNodeParser handles the `any` keyword. It also claims the `symbol`
// keyword, but SymbolTypeNodeParser is registered earlier in the chain so
// symbols become SymbolType (src/NodeParser/AnyTypeNodeParser.ts).
type AnyTypeNodeParser struct{}

func NewAnyTypeNodeParser() *AnyTypeNodeParser {
	return &AnyTypeNodeParser{}
}

func (p *AnyTypeNodeParser) SupportsNode(node *ast.Node) bool {
	return node.Kind == ast.KindAnyKeyword || node.Kind == ast.KindSymbolKeyword
}

func (p *AnyTypeNodeParser) CreateType(_ *ast.Node, _ *Context, _ *types.ReferenceType) types.Type {
	return &types.AnyType{}
}
